core/src/executor/select/mod.rs

```diff
@@ -287,15 +287,23 @@ pub async fn select_with_labels<'a>(
     });
 
     let rows = aggregate.apply(rows).await?;
-    let rows = sort
-        .apply(rows)
-        .await?
-        .and_then(move |(aggregated, context)| {
-            let blend = Rc::clone(&blend);
+    let rows = rows.and_then(move |aggregate_context| {
+        let blend = Rc::clone(&blend);
+        let AggregateContext { aggregated, next } = aggregate_context;
+        let rc_aggregated = aggregated.clone().map(Rc::new);
+
+        async move {
+            let row = blend.apply(rc_aggregated.clone(), Rc::clone(&next)).await?;
 
-            async move { blend.apply(aggregated, context).await }
-        });
+            Ok((AggregateContext { aggregated, next }, row))
+        }
```

Comment:
Currently `sort.rs` apply method takes `AggregateContext` as a param.
<img width="467" alt="image" src="https://user-images.githubusercontent.com/2025065/188304925-1d80404b-9a88-4c16-897b-1383d041b050.png">

As we do in `blend.rs` apply method,
<img width="398" alt="image" src="https://user-images.githubusercontent.com/2025065/188304943-03202272-69fd-4b42-9865-76ac7172e6d5.png">

It looks that if we change the `Sort::apply` the types of input params similar with `Blend::apply`, then we might not need to clone aggregated in here.